### PR TITLE
Documenting MBM2/BBB Ethernet Support + examples

### PR DESCRIPTION
**Pull request by CatGarab**
Documenting the ethernet port that's now available for use for the Pumpkin MBM2 and Beaglebone Black targets.

Also adding example programs for sending and receiving messages using TCP.

### CODE REVIEW

**Comment by CatGarab** [comment]
[Pumpkin ethernet](https://trello.com/c/O0ctR98S/16-pumpkin-ethernet)

File: docs/working-with-the-mbm2.rst [hunk 1]
```diff
@@ -204,36 +204,128 @@ as long as they have not already been assigned to another peripheral.
...
+    
+    /* Set the pin's value */
+    fd = open("/sys/class/gpio/gpio45/value", O_WRONLY);
+    write(fd, &value, 1);
+    close(fd);
+    
+    /* Read the value back */
+    fd = open("/sys/class/gpio/gpio45/value", O_RDONLY);
+    char strValue[3];
+    write(fd, strValue, 1);
```
Comment (plauche) [inline docs/working-with-the-mbm2.rst]: Should this `write` be a `read`?

File: docs/working-with-the-bbb.rst [hunk 2]
```diff
@@ -371,6 +371,43 @@ An example program might look like this:
...
+- `SCTP tutorial <http://petanode.com/blog/posts/introduction-to-the-sctp-socket-api-in-linux.html>`__
+- `Packet Sender <https://packetsender.com/>`__ - A tool to send test packets between an OBC and a host computer
+
+.. note:: By default, Windows Firewall will block many incoming packet types. This may impact testing.
+
+Configuration
+^^^^^^^^^^^^^
+
+The static IP address can be updated by editing the `/etc/network/interfaces` file.
+By default the address is ``168.168.2.20``.
```
Comment (jacoffey3) [inline docs/working-with-the-bbb.rst]: Why this address?

Comment (CatGarab) [reply docs/working-with-the-bbb.rst]: On the host side, the general recommendation seemed to be to make the network component (the first number) of the local ethernet IP different from your actual network IP. Mine was 192.168.*.*, so I went with 168.168.*.* instead. Beyond that it's completely arbitrary.

**Pull request merged**